### PR TITLE
Automatic Python release workflow

### PR DESCRIPTION
**Pull request by woodruffw**
Per #22: this is a demonstration of how we can mostly automate our releases, in this case for Python.

cc @loosebazooka @znewman01

### CODE REVIEW

**Comment by woodruffw** [comment]
N.B.: This workflow won't function yet, since I haven't created a PyPI API token for `sigstore-protobuf-specs` yet.

**Comment by woodruffw** [comment]
This just provides the basic workflow and scaffolding needed to perform a release automatically.

We probably also want additional scaffolding (e.g. a top level `make release-python` target), but I'll hold off on anything like that until this passes an initial sniff test 🙂

**Comment by znewman01** [comment]
> ~N.B.: This workflow won't function yet, since I haven't created a PyPI API token for `sigstore-protobuf-specs` yet.~
> 
> Edit: created, waiting for it to be added to the repo.

Done: 
Screenshot 2022-12-07 at 3 27 36 PM

Hopefully that worked, let me know if not

**Comment by woodruffw** [comment]
> Hopefully that worked, let me know if not

I don't have merge or push access, so someone else will have to give this a spin 🙂

File: .github/workflows/python-release.yml
Comment (woodruffw) [reply .github/workflows/python-release.yml:21]: Yes, thanks for catching that!

**Comment by woodruffw** [comment]
If one of the maintainers merges this, you _should_ be able to push out a release as-is with:

```
git tag release/python/v0.0.1a1
git push origin release/python/v0.0.1.a1
```

(Since that's the current pending version for the Python package).

**Comment by znewman01** [comment]
I will give that a go soon, thanks

**Review by znewman01 — APPROVED** [review]
> If one of the maintainers merges this, you should be able to push out a release as-is with:

Textbook social engineering, good work @woodruffw

**Comment by znewman01** [comment]
Close, but no cigar: https://github.com/sigstore/protobuf-specs/actions/runs/3653257167/jobs/6172519309

File: .github/workflows/python-release.yml
Comment (znewman01) [inline .github/workflows/python-release.yml:20]: YAML!

You need `run: |` on this line

Comment (woodruffw) [reply .github/workflows/python-release.yml:20]: 🤦

**Pull request merged**
